firedrake/matrix.py

```diff
@@ -34,9 +34,9 @@ def __init__(self, a, bcs, mat_type):
         # (so we can't use a set, since the iteration order may differ
         # on different processes)
 
-        ufl.Matrix.__init__(self, test.function_space(), trial.function_space())
-        # Define arguments after `Matrix.__init__` since BaseForm sets `self._arguments` to None
-        self._arguments = arguments
+        ufl.Matrix.__init__(self, test.ufl_function_space(), trial.ufl_function_space())
```

Comment:
Are there reasons to use `ufl_function_space()` over `function_space()`?

Reply:
No good spot. That was from when I was debugging the `_arguments` issue. Will revert.